### PR TITLE
Implement code generation for interface and enum

### PR DESCRIPTION
**Pull request by 0xSalman**
Fixes https://github.com/DealTap/graphql-gen-go/issues/1

Test notes will be added in later PRs once the code base is stable

### CODE REVIEW

**Comment by 0xSalman** [comment]
@elau-dt should we address some of the `FIXME` (code refactoring) or let it be for now?

File: generator/generator.go [hunk 1]
```diff
@@ -85,13 +85,16 @@ type FieldDef struct {
   Description string
   Type        *Typ
   gqlField    *introspection.Field
+  Args        []*FieldDef
 }
 
 func NewField(t *introspection.Field) (fld *FieldDef) {
+
```
Comment (edlau) [inline generator/generator.go:92]: Drop space line?

File: generator/generator.go [hunk 2]
```diff
@@ -421,12 +483,14 @@ func (g Generator) GenSchemaResolversFile() ([]byte, []*TypeDef) {
   g.P("")
   g.P("import (")
   g.In()
-  g.P(`"time"`)
+  // FIXME include only when time field is present
+  //g.P(`"time"`)
   g.P(`graphql "github.com/neelance/graphql-go"`)
```
Comment (edlau) [inline generator/generator.go]: (Non blocking) Do we need another FIXME to make this link outside `generator.go`?

**Comment by elau-dt** [comment]
@salmana1 defer FIXME for now I would say.

File: generator/generator.go (hunk 1 above)
Comment (0xSalman) [reply generator/generator.go:92]: this function is more than 10 lines so I think a line after the function signature improves readability.

File: generator/generator.go (hunk 2 above)
Comment (0xSalman) [reply generator/generator.go]: sure - I will do another reading of protoc-gen-go's code to see how they are handling these imports

**Review by edlau — APPROVED** [review]
Visually inspected.

**Pull request merged**